src/App.jsx

```diff
@@ -14,33 +16,96 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPrapedGoods(goods, sortField, reversed) {
+  let prepearedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === SORT_FIELD_ALPHABET) {
+    prepearedGoods.sort();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === SORT_FIELD_LENGTH) {
+    prepearedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (reversed) {
+    prepearedGoods.reverse();
+  }
+
+  if (sortField === '') {
+    prepearedGoods = [...goodsFromServer];
+  }
+
+  if (reversed && sortField === '') {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPrapedGoods(goodsFromServer, sortField, reversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function resetFieldAndRevrsed() {
```

Comment:
There is a typo in the function name `resetFieldAndRevrsed`. It should be `resetFieldAndReversed` to accurately reflect its purpose.